src/index.d.ts

```diff
@@ -85,76 +85,76 @@ declare namespace admin.app {
 
 declare namespace admin.auth {
   interface UserMetadata {
-    lastSignInTime: string;
-    creationTime: string;
+    readonly lastSignInTime: string;
+    readonly creationTime: string;
 
     toJSON(): Object;
   }
 
   interface UserInfo {
-    uid: string;
-    displayName: string;
-    email: string;
-    phoneNumber: string;
-    photoURL: string;
-    providerId: string;
+    readonly uid: string;
+    readonly displayName: string;
+    readonly email: string;
+    readonly phoneNumber: string;
+    readonly photoURL: string;
+    readonly providerId: string;
 
     toJSON(): Object;
   }
 
   interface UserRecord {
-    uid: string;
-    email?: string;
-    emailVerified: boolean;
-    displayName?: string;
-    phoneNumber?: string;
-    photoURL?: string;
-    disabled: boolean;
-    metadata: admin.auth.UserMetadata;
-    providerData: admin.auth.UserInfo[];
-    passwordHash?: string;
-    passwordSalt?: string;
-    customClaims?: Object;
-    tokensValidAfterTime?: string;
+    readonly uid: string;
+    readonly email?: string;
+    readonly emailVerified: boolean;
+    readonly displayName?: string;
+    readonly phoneNumber?: string;
+    readonly photoURL?: string;
+    readonly disabled: boolean;
+    readonly metadata: admin.auth.UserMetadata;
+    readonly providerData: admin.auth.UserInfo[];
+    readonly passwordHash?: string;
+    readonly passwordSalt?: string;
+    readonly customClaims?: Object;
+    readonly tokensValidAfterTime?: string;
 
     toJSON(): Object;
   }
 
   interface UpdateRequest {
```

Comment:
Can't make this readonly. Developers should be able to incrementally construct an UpdateRequest if they wish to.

```
const  req: UpdateRequest = {};
req.uid = 'alice'; // This will fail if uid is readonly
```